api/woeip/apps/air_quality/views.py

```diff
@@ -2,40 +2,99 @@
-        else:
-            messages.add_message(self.request, messages.ERROR, 'File upload error')
-            return render(self.request, 'air_quality/upload.html', {
-                'form': CollectionForm
-            })
-
-
-class ViewCollection(View):
-    """Provide temporary development page to view all uploaded SessionDatas."""
-    def get(self, request):
-        collection_list = Collection.objects.all()
-        return render(self.request, 'air_quality/view.html', {
-            'collection_list': collection_list
-        })
+class CalibrationViewSet(viewsets.ModelViewSet):
+    queryset = models.Calibration.objects.all()
+    serializer_class = serializers.CalibrationSerializer
+
+    def get_queryset(self):
+        queryset = models.Calibration.objects.all()
+
+        return queryset
+
+
+class CollectionViewSet(viewsets.ModelViewSet):
+    queryset = models.Collection.objects.all()
+    serializer_class = serializers.CollectionSerializer
+
+    @action(detail=True, methods=["GET"])
+    def geo(self, request, pk=None):
+        collection = get_object_or_404(self.queryset, pk=pk)
+        pollutant_values = models.PollutantValue.objects.filter(
+            collection_file__collection=collection,
+        )
+        serializer = serializers.CollectionGeoSerializer(
+            {
+                "metadata": collection,
+                "pollutant_values": pollutant_values
+            },
+            context={"request": request},
+        )
+
+        return Response(serializer.data)
+
+
+class CollectionFileViewSet(viewsets.ModelViewSet):
+    queryset = models.CollectionFile.objects.all()
+    serializer_class = serializers.CollectionFileSerializer
+
+    def get_queryset(self):
+        queryset = models.CollectionFile.objects.all()
+
+        return queryset
+
+
+class DeviceViewSet(viewsets.ModelViewSet):
+    queryset = models.Device.objects.all()
+    serializer_class = serializers.DeviceSerializer
+
+    def get_queryset(self):
+        queryset = models.Device.objects.all()
+        device_name = self.request.query_params.get("name", None)
+        if device_name:
+            queryset = queryset.filter(name=device_name)
+
+        serial = self.request.query_params.get("serial", None)
+        if serial:
+            queryset = queryset.filter(serial=serial)
+
+        return queryset
+
+
+class PollutantViewSet(viewsets.ModelViewSet):
+    queryset = models.Pollutant.objects.all()
+    serializer_class = serializers.PollutantSerializer
+
+
+class PollutantValueViewSet(viewsets.ModelViewSet):
+    queryset = models.PollutantValue.objects.all()
+    serializer_class = serializers.PollutantValueSerializer
+
+
+class SensorViewSet(viewsets.ModelViewSet):
+    queryset = models.Sensor.objects.all()
+    serializer_class = serializers.SensorSerializer
+
+    def get_queryset(self):
+        queryset = models.Sensor.objects.all()
+        sensor_name = self.request.query_params.get("name", None)
```

Comment:
Let's be consistent about using `'` vs `"` across the codebase. It looks like there are slightly more `'` so I would lean towards sticking to that for simplicity (and `"""` for docstrings), but no strong preference aside from consistency!

Reply:
Will be resolved in #143 so I'll leave it for now